### PR TITLE
poezio: update to 0.14, switch to Python 3.9

### PR DESCRIPTION
**Pull request by OscarL**
Should be merged after #8035 (done) and #8036 (done, `slixmpp` will need a revbump for rebuild first).

Tested on beta4 64 and 32 bits. Was able to auto-connect to the default server and get an ID. (no further testing perfomed).

### CODE REVIEW

File: net-im/poezio/poezio-0.14.recipe [hunk 1]
```diff
@@ -29,36 +29,37 @@ PROVIDES="
...
-	aiodns_python3
-	pyasn1_modules_python3
-	pyasn1_python3
-	pycares_python3
-	slixmpp_python3 >= 1.3.0
-	cmd:python3
+	aiodns_python39
+	pyasn1_modules_python39
+	pyasn1_python39
+	pycares${secondaryArchSuffix}_python39
```
Comment (korli) [inline net-im/poezio/poezio-0.14.recipe:35]: pycares_x86_python39 should also provide pycares_python39

```suggestion
	pycares_python39
```

File: net-im/poezio/poezio-0.14.recipe [hunk 2]
```diff
@@ -29,36 +29,37 @@ PROVIDES="
...
+	slixmpp${secondaryArchSuffix}_python39 >= 1.8.2
```
Comment (korli) [inline net-im/poezio/poezio-0.14.recipe:36]: is slixmpp really x86?

```suggestion
	slixmpp_python39 >= 1.8.2
```

File: net-im/poezio/patches/poezio-0.14.patchset [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+-    '/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt',
++    '/boot/system/data/ssl/CARootCertificates.pem',
++    '/boot/system/data/ssl/cert.pem',
++    '/boot/system/data/ssl/certs/ca-certificates.crt',
++    '/boot/system/data/ssl/certs/ca-bundle.crt',
++    '/boot/system/data/pki/tls/certs/ca-bundle.crt',
++    '/boot/system/data/ssl/certs/ca-certificates.crt',
++    '/boot/system/data/ca-certificates/extracted/tls-ca-bundle.pem',
++    '/boot/system/data/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt',
+ }
```
[the remark this answers is not in the document]
Comment (korli) [reply net-im/poezio/patches/poezio-0.14.patchset:93]: @Begasus in a patch it's a bit difficult. Adding the needed line should be OK

[the remark this answers is not in the document]
Comment (Begasus) [reply net-im/poezio/patches/poezio-0.14.patchset:93]: I know @korli , hence I mentioned `(in the recipe)`, but it's fine this way too (if it works) :)

[the remark this answers is not in the document]
Comment (korli) [reply net-im/poezio/patches/poezio-0.14.patchset:93]: @pulkomandy IMO in this case the filename and path are Haiku specfic. it should be some build config instead, no way to use find_path here.

**Pull request merged**